plugins/inputs/prometheus/kubernetes.go

```diff
@@ -0,0 +1,211 @@
+package prometheus
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net"
+	"net/url"
+	"os/user"
+	"path/filepath"
+	"sync"
+	"time"
+
+	"github.com/ericchiang/k8s"
+	corev1 "github.com/ericchiang/k8s/apis/core/v1"
+	"github.com/ghodss/yaml"
+)
+
+type payload struct {
+	eventype string
+	pod      *corev1.Pod
+}
+
+// loadClient parses a kubeconfig from a file and returns a Kubernetes
+// client. It does not support extensions or client auth providers.
+func loadClient(kubeconfigPath string) (*k8s.Client, error) {
+	data, err := ioutil.ReadFile(kubeconfigPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed reading '%s': %s", kubeconfigPath, err.Error())
+	}
+
+	// Unmarshal YAML into a Kubernetes config object.
+	var config k8s.Config
+	if err := yaml.Unmarshal(data, &config); err != nil {
+		return nil, err
+	}
+	return k8s.NewClient(&config)
+}
+
+func start(p *Prometheus) error {
+	client, err := k8s.NewInClusterClient()
+	if err != nil {
+		u, err := user.Current()
+		if err != nil {
+			return fmt.Errorf("Failed to get current user - %s", err.Error())
+		}
+		configLocation := filepath.Join(u.HomeDir, ".kube/config")
+		if p.KubeConfig != "" {
+			configLocation = p.KubeConfig
+		}
+		client, err = loadClient(configLocation)
+		if err != nil {
+			return err
+		}
+	}
+
+	p.ctx, p.cancel = context.WithCancel(context.Background())
+	p.wg = sync.WaitGroup{}
+	in := make(chan payload)
+
+	p.wg.Add(1)
+	go func() {
+		defer p.wg.Done()
+		t := time.NewTimer(time.Second)
+		for {
+			select {
+			case <-p.ctx.Done():
+				t.Stop()
+				log.Printf("I! [inputs.prometheus] shutting down")
+				return
+			case <-t.C:
```

Comment:
Just remembered to take a look at the retry...  I think you will need to move the timer into the for loop so that you get a new time on each try.